custom/crediblex/portfolio/loanaccount/src/main/java/com/crediblex/fineract/portfolio/loanaccount/service/CustomLoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -659,4 +680,173 @@ public CommandProcessingResult makeLoanRepayment(final LoanTransactionType repay
         return result;
     }
 
+    public CommandProcessingResult makeLoanRepaymentWithChargeRefundChargeType(final LoanTransactionType repaymentTransactionType,
+                                                                               final Long loanId, final JsonCommand command, final boolean isRecoveryRepayment, final String chargeRefundChargeType) {
+        final LoanRepaymentsSummaryMapper loanRepaymentSummaryMapper = new LoanRepaymentsSummaryMapper();
+        final String loanRepaymentSummarySql = loanRepaymentSummaryMapper.loanPaymentsSummarySchema();
+        final Collection<LoanSchedulePeriodData> loanSchedulePeriods = this.jdbcTemplate.query(loanRepaymentSummarySql, loanRepaymentSummaryMapper, loanId);
+
+        final LoanCurrencyDataMapper loanCurrencyDataMapper = new LoanCurrencyDataMapper();
+        final String loanCurrencySql = loanCurrencyDataMapper.loanPaymentsSummarySchema();
+        final CurrencyData currency = this.jdbcTemplate.queryForObject(loanCurrencySql, loanCurrencyDataMapper, loanId);
```

Comment:
When you fetch a loan object it wil have a currency data, you dont need to fetch one here